stream/stream.c

```diff
@@ -803,7 +803,7 @@ int stream_skip_bom(struct stream *s)
 struct bstr stream_read_complete(struct stream *s, void *talloc_ctx,
                                  int max_size)
 {
-    if (max_size > 1000000000)
+    if (max_size <= 0)
```

Comment:
there is an integer overflow in this function when called with `max_size=INT_MAX`

the limit really was there for a reason and it shouldn't be removed

Reply:
> there is an integer overflow in this function when called with max_size=INT_MAX

ah, it adds forcefully a 1 byte padding, fixed.

> the limit really was there for a reason and it shouldn't be removed

There is no reason for arbitrary limit like that.